weave/ecosystem/wandb/panel_time_series.py

```diff
@@ -197,35 +197,54 @@ def initialize(self) -> TimeSeriesConfig:
             input_node.type, weave.types.optional(weave.types.Timestamp())
         )
 
-        min_x_called = col_fn.val.min()
+        item_fn = weave_internal.const(
+            item_fn.val,
+            item_fn.type,
+        )
+
+        min_x_called = weave_internal.better_call_fn(col_fn, input_node).min()  # type: ignore
         min_x = weave_internal.const(
             min_x_called,
             weave.types.Function(col_fn.type.input_types, min_x_called.type),  # type: ignore
         )
 
-        max_x_called = col_fn.val.max()
+        max_x_called = weave_internal.better_call_fn(col_fn, input_node).max()  # type: ignore
         max_x = weave_internal.const(
             max_x_called,
             weave.types.Function(col_fn.type.input_types, max_x_called.type),  # type: ignore
         )
 
         mark = weave_internal.const("bar")
 
+        label_node = first_column_of_type(
+            input_node.type,
+            weave.types.optional(
+                weave.types.union(
+                    weave.types.String(),
+                    weave.types.Boolean(),
+                )
+            ),
+        )[1]
+
+        label_node = weave_internal.const(
+            label_node.val,
+            label_node.type,
+        )
+
+        agg_node = weave_internal.define_fn(
+            {"group": input_node.type},  # type: ignore
+            lambda group: group.count(),
+        )
+
+        agg = weave_internal.const(
```

Comment:
Why did you need to wrap in another const node here?

Did you check if you can still configure panel_time_series from the UI?

Reply:
Oh - so we need all these to be RuntimeConstNodes, not just ConstNodes. This might just be an artifact of testing flow, but doing something like this (which mimics `initialize`):
```
    init_config_node = panel.initialize()
    init_config = weave.use(init_config_node)
    panel.config = init_config
    render_node = panel.render()
    res = weave.use(render_node)
```

The `render` op gets called with these nodes and internally expects Runtime nodes.

If this is a pattern violation, happy to change

Reply:
I totally missed that you were just rewrapping these, I thought you were double wrapping...

Well, the one thing about these plots is I was trying to have them read really well, like be implemented as we would want a user to do.

Having to change nodes to dispatch nodes is no good, also using better_call_fn instead of the __call__ dunder.

Reply:
fixed (: Made define_fn return runtime node